File: src/main/java/com/sidgate/poker/FullHouse.java
package com.sidgate.poker;

import com.sidgate.deck.Hand;

public class FullHouse extends HandType {

	public FullHouse(HandType next) {
		super(next);
	}
	@Override
	protected boolean process(Hand hand) {
		return hand.cardsGroupedByRank().stream().anyMatch(list -> list.size() == 3)
			&& hand.cardsGroupedByRank().stream().anyMatch(list -> list.size() == 2);
	}

	@Override
	public String getName() {
		return "full-house";
	}

}

File: src/main/kotlin/sidgate/poker/FullHouse.kt
package sidgate.poker

import sidgate.deck.Hand

class FullHouse(next: HandType?) : HandType(next) {
    override fun process(hand: Hand) =
            (hand.cardsGroupedByRank().any { it.size == 3 } && hand.cardsGroupedByRank().any { it.size == 2 })

    override val name: String
        get() = "full-house"
}